express/features/support/steps/client-steps.ts

```diff
@@ -30,3 +31,9 @@ Given("the user submits a valid public key without headers", async function () {
     await enterTextIntoTextInput(this.page, VALID_PUBLIC_KEY_WITHOUT_HEADERS, "serviceUserPublicKey");
     await clickSubmitButton(this.page);
 });
+
+Then("they should see the public key they just entered in the textarea", async function () {
+    const textArea = await this.page.$("#serviceUserPublicKey");
+    const textAreaContent = await this.page.evaluate((element: {textContent: string}) => element.textContent, textArea);
+    assert.equal(textAreaContent, VALID_PUBLIC_KEY_WITHOUT_HEADERS.replace(/\n/g, "").replace(/\+/g, " "));
```

Comment:
Why do the `+`s need to be replaced with spaces?
They are valid Base64 characters ([spec](https://www.rfc-editor.org/rfc/rfc4648.txt)). Does the library we use complain about them?

Reply:
This was a late in the day brain fart from me - URL encoding must make Express think the +s are actually spaces when you retrieve the query parameter and in my head the Base64 + was interchangeable with a space 🤦🏻‍♂️ 

Anyyywayyyy, it puts the correct text in the text area now so that other than the newlines which still need removing, we don't rejig the sample text.

Reply:
TIL `encodeUrlComponent` is what I actually wanted long ago.